lms/resources/_js_config/__init__.py

```diff
@@ -118,14 +118,29 @@ def enable_oauth2_redirect_error_mode(
 
         :param error_details: Technical details of the error
         :type error_details: str
-        :param auth_url: URL for the "Try again" button in the dialog
-        :type auth_url: str
+        :param auth_route: route for the "Try again" button in the dialog
+        :type auth_route: str
         :param is_scope_invalid: `True` if authorization failed because the
           OAuth client does not have access to all the necessary scopes
         :type is_scope_invalid: bool
         :param canvas_scopes: List of scopes that were requested
         :type canvas_scopes: list(str)
         """
+        if self._lti_user:
+            auth_url = self._request.route_url(
+                auth_route,
+                _query=[
+                    (
+                        "authorization",
+                        BearerTokenSchema(self._request).authorization_param(
+                            self._lti_user
+                        ),
+                    )
+                ],
+            )
```

Comment:
```suggestion
            bearer_token = BearerTokenSchema(self._request).authorization_param(
                self._lti_user
            )
            auth_url = self._request.route_url(
                auth_route, _query=[("authorization", bearer_token)]
            )
```

I've noticed sometimes when black really spaces things out like this it can actually help make it more readable and shorter to pull a bit out. It's technically not a one liner any more, but if it gets really stringy like this I think it can help.

I've seen this help with some of the very nested `h_matchers` assertions as well.

Reply:
:+1: Done